### PR TITLE
README: clarify non-goal and why

### PR DESCRIPTION
**Pull request by magodo**
Clarify non-goal and why in the README.md.

### CODE REVIEW

**Review by ms-henglu — APPROVED** [review]
LGTM! 🎉

File: README.md [hunk 1]
```diff
@@ -104,13 +108,25 @@ This means if the output directory has an active Terraform workspace, i.e. there
...
+
+### AzureRM Provider Validation
+
+When generating the Terraform configuration, not all properties of the resource are exported for different reasons.
+
+One reason is because there are flexible cross-property constraints defined in the AzureRM Terraform provider. E.g. `property_a` conflits with `property_b`. This might due to the nature of the API, or might be due to some deprecation process of the provider (e.g. `property_a` is deprecated in favor of `property_b`, but kept for backwards compatibility). These constraints require some properties must be absent in the Terraform configuration, otherwise, the configuration is not a valid and will fail during `terraform validate`.
+
+Another reason is that an Azure resource can be a property of its parent resource (e.g. `azurerm_subnet` can be its own resource, or be a property of `azurerm_virtual_network`). Per Terraform's best practice, users should only use one of the forms, not both. `aztfy` chooses to always generate all the resources, but omit the property in the parent resource that represents to the child resource.
```
Comment (mybayern1974) [inline README.md:129]:
```suggestion
Another reason is that an Azure resource can be a property of its parent resource (e.g. `azurerm_subnet` can be its own resource, or be a property of `azurerm_virtual_network`). Per Terraform's best practice, users should only use one of the forms, not both. `aztfy` chooses to always generate all the resources, but omit the property in the parent resource that represents the child resource.
```

File: README.md [hunk 2]
```diff
@@ -104,13 +108,25 @@ This means if the output directory has an active Terraform workspace, i.e. there
...
 
-Currently, this tool only works on the assumption that there is 1:1 mapping between Azure resources and the Terraform resources.
+Currently, this tool only works on the assumption that there is 1:1 mapping between Azure resources and the Terraform resources. For those property-like Terraform resources, `aztfy` will just ignore them.
+
+### AzureRM Provider Validation
+
+When generating the Terraform configuration, not all properties of the resource are exported for different reasons.
+
+One reason is because there are flexible cross-property constraints defined in the AzureRM Terraform provider. E.g. `property_a` conflits with `property_b`. This might due to the nature of the API, or might be due to some deprecation process of the provider (e.g. `property_a` is deprecated in favor of `property_b`, but kept for backwards compatibility). These constraints require some properties must be absent in the Terraform configuration, otherwise, the configuration is not a valid and will fail during `terraform validate`.
```
Comment (mybayern1974) [inline README.md:127]: The paragraph of the Reason One may introduce ambiguous meanings: 1) no properties would be generated at all 2) only one of those conflicted properties would be generated. If it were the 2nd case a further ask would be raised: which one would be generated and why choose that one. Given assume the fact is the 1st case, suggest tune the description a bit.

**Comment by mybayern1974** [comment]
Left some comments but overall LGTM!

File: README.md (hunk 2 above)
Comment (magodo) [reply README.md:127]: The way how we handle this case might keep changing. Here my intent is only to tell the reason, but not provide a commit about the implementation details. Instead, we only commit that our goal is to pass `terraform plan`.

**Pull request merged**